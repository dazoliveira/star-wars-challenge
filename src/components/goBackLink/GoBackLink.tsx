import { useCallback } from "react";
import { Box, Link, Typography } from "@material-ui/core";
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import { useStyles } from "./styles";
import { history } from "utils";

export type GoBackLinkProps = {
    variant: boolean
    url: string
}

export default function GoBackLink({ variant, url }: GoBackLinkProps) {

    const classes = useStyles({ variant })

    const onClickHandle = useCallback(() => {
        history.push(url)
    }, [url])
    
    return (
        <Box className={classes.root}>
            <Link onClick={onClickHandle}>
                <ArrowBackIcon fontSize='inherit' color='secondary'/>
                <Typography  variant='body1' component='div'>back</Typography>
            </Link>
        </Box>    
    )
}