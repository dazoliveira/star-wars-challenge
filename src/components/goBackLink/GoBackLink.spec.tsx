import { screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event';
import { renderWithTheme } from "utils/tests/helpers";
import GoBackLink, { GoBackLinkProps } from './GoBackLink';


const props: GoBackLinkProps = {
    variant: false,
    url: '/welcome'
}

describe('<GoBackLinkProps />', () => {
    it('should render back link', () => {
        const { container } = renderWithTheme(<GoBackLink {...props} />)
        expect(screen.getByText('back')).toBeInTheDocument()

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should back to /welcome url on back click', async () => {
       const { getByText } = renderWithTheme(<GoBackLink {...props} />)

        const link = getByText(/back/i)

        userEvent.click(link)

        const location = window.location.pathname
        const welcomePage = '/welcome'

        expect(location).toEqual(welcomePage)
    })
});