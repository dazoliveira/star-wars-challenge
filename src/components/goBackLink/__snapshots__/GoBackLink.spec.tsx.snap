// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoBackLinkProps /> should render back link 1`] = `
<div
  class="MuiBox-root MuiBox-root-3 makeStyles-root-1 makeStyles-root-2"
>
  <a
    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeInherit"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
      />
    </svg>
    <div
      class="MuiTypography-root MuiTypography-body1"
    >
      back
    </div>
  </a>
</div>
`;
